compliance/collection/intervals/intervals.go

```diff
@@ -0,0 +1,71 @@
+package intervals
+
+import (
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	// Used for testing purposes, to mock random values.
+	randFloat64 = rand.Float64
+)
+
+// NodeScanIntervals generates node scanning intervals using randomized values.
+type NodeScanIntervals struct {
+	base       time.Duration
+	deviation  float64
+	initialMax time.Duration
+}
+
+// deviateDuration randomly deviates a duration by a given percentage. Example:
+// duration of 10s with 5% deviation means a random duration between 5s and 15s.
+func deviateDuration(d time.Duration, percentage float64) time.Duration {
+	min, max := 1.0-percentage, 1.0+percentage
+	dev := randFloat64()*(max-min) + min
+	return multiplyDuration(d, dev)
+}
+
+// multiplyDuration multiplies a duration by a float64 and returns the resulting
+// duration.
+func multiplyDuration(d time.Duration, factor float64) time.Duration {
+	return time.Duration(float64(time.Second) * d.Seconds() * factor)
+}
+
+// NewNodeScanIntervalFromEnv creates node scanning intervals from environment
+// variables.
+func NewNodeScanIntervalFromEnv() NodeScanIntervals {
+	i := NodeScanIntervals{}
+	i.base = env.NodeScanningInterval.DurationSetting()
+	i.deviation = 0.0
+	if env.NodeScanningIntervalDeviation.IntegerSetting() > 0 {
+		d := math.Min(float64(env.NodeScanningIntervalDeviation.IntegerSetting()), 100.0)
+		i.deviation = d / 100.0
+	}
+	i.initialMax = env.NodeScanningMaxInitialWait.DurationSetting()
+	log.Infof("scanning intervals: base=%s deviation=%.2f initialMax=%s",
+		i.base, i.deviation, i.initialMax)
```

Comment:
Let's make it a bit more user friendly

```suggestion
	log.Infof("Scanning intervals: base interval=%s deviation from base=±%s, first scan in maximally=%s",
		i.base, multiplyDuration(i.base, i.deviation).String(), i.initialMax)
```

Reply:
@vikin91:

> Let's make it a bit more user friendly

Not sure what the goal is here; I know logs are customer-facing, but do we want users to be able to evaluate the logs and understand them without additional context what those log lines mean?

Reply:
My goal is that our future us would understand this when debugging or analyzing a bundle with logs from a customer.
I am a bit worried about the term deviation as it may be easily confused with standard deviation (square difference), whereas here, we use absolute deviation. Adding the `±` symbol could help to guide the person reading this into treating this as a _range_ instead of _standard deviation_ as both of them are measures of [statistical dispersion](https://en.wikipedia.org/wiki/Statistical_dispersion).